src/Console.php

```diff
@@ -0,0 +1,249 @@
+     * will use this to automatically schedule their own callback
+     * and allowing you a cleaner syntax, such as.
+     *
+     * $console->setModel($user, 'generateReport');
+     *
+     * @var bool
+     */
+    protected $sseInProgress = false;
+
+    /**
+     * Stores object jsSSE which is used for communication.
+     *
+     * @var jsSSE
+     */
+    public $sse;
+
+    /**
+     * Supply callback which will be executed in "background" sending
+     * your console output.
+     *
+     * @param $callback string
+     *
+     * @return $this
+     */
+    public $_output_bypass = false;
+
+    /**
+     * Set a callback method which will be executed with the output sent back to the terminal.
+     *
+     * Argument passed to your callback will be $this Console. You may perform calls
+     * to methods such as
+     *
+     *   $console->output()
+     *   $console->outputHTML()
+     *
+     * If you are using setModel, and if your model implements atk4\core\DebugTrait,
+     * then you you will see debug information generated by $this->debug() or $this->log().
+     *
+     * This intercepts default application logging for the duration of the process.
+     *
+     * If you are using runCommand, then server command will be executed with it's output
+     * (STDOUT and STDERR) redirected to the console.
+     *
+     * While inside a callback you may execute runCommand or setModel multiple times.
+     */
+    public function set($callback)
+    {
+        $this->sse = $this->add('jsSSE');
+        $this->sse->set(function () use ($callback) {
+            $this->sseInProgress = true;
+
+            try {
+                ob_start(function ($content) {
+                    if ($this->_output_bypass) {
+                        return $content;
+                    }
+
+                    $output = '';
+                    $this->sse->echoFunction = function ($str) use (&$output) {
+                        $output .= $str;
+                    };
+                    $this->output($content);
+                    $this->sse->echoFunction = false;
+
+                    return $output;
+                }, 2);
+
+                call_user_func($callback, $this);
+            } catch (\atk4\core\Exception $e) {
+                $lines = explode("\n", $e->getHTMLText());
+
+                foreach ($lines as $line) {
+                    $this->outputHTML($line);
+                }
+            } catch (\Error $e) {
+                $this->output('Error: '.$e->getMessage());
+            } catch (\Exception $e) {
+                $this->output('Exception: '.$e->getMessage());
+            }
+            $this->sseInProgress = false;
+        });
+        $this->js(true, $this->sse);
+
+        return $this;
+    }
+
+    /**
+     * Output a single line to the console.
+     *
+     * @param $text string
+     *
+     * @return $this
+     */
+    public function output($text)
+    {
+        $this->_output_bypass = true;
```

Comment:
how about simply using `return $this->outputHTML(htmlspecialchars($text));` instead in this method to avoid code duplication?